crates/esplora/src/blocking_ext.rs

```diff
@@ -112,6 +164,235 @@ impl EsploraExt for esplora_client::BlockingClient {
+                                let tx_count = txs.len();
+                                last_seen = txs.last().map(|tx| tx.txid);
+                                spk_txs.extend(txs);
+                                if tx_count < 25 {
+                                    break Ok((spk_index, spk_txs));
+                                }
+                            }
+                        }
+                    })
+                })
+                .collect::<Vec<JoinHandle<Result<TxsOfSpkIndex, Error>>>>();
+
+            if handles.is_empty() {
+                break;
+            }
+
+            for handle in handles {
+                let (index, txs) = handle.join().expect("thread must not panic")?;
+                last_index = Some(index);
+                if !txs.is_empty() {
+                    last_active_index = Some(index);
+                }
+                for tx in txs {
+                    let _ = tx_graph.insert_tx(tx.to_tx());
+                    if let Some(anchor) = anchor_from_status(&tx.status) {
+                        let _ = tx_graph.insert_anchor(tx.txid, anchor);
+                    }
+
+                    let previous_outputs = tx.vin.iter().filter_map(|vin| {
+                        let prevout = vin.prevout.as_ref()?;
+                        Some((
+                            OutPoint {
+                                txid: vin.txid,
+                                vout: vin.vout,
+                            },
+                            TxOut {
+                                script_pubkey: prevout.scriptpubkey.clone(),
+                                value: Amount::from_sat(prevout.value),
+                            },
+                        ))
+                    });
+
+                    for (outpoint, txout) in previous_outputs {
+                        let _ = tx_graph.insert_txout(outpoint, txout);
+                    }
+                }
+            }
+
+            let last_index = last_index.expect("Must be set since handles wasn't empty.");
+            let gap_limit_reached = if let Some(i) = last_active_index {
+                last_index >= i.saturating_add(stop_gap as u32)
+            } else {
+                last_index + 1 >= stop_gap as u32
+            };
+            if gap_limit_reached {
+                break;
+            }
+        }
+
+        Ok(last_active_index)
+    }
+
+    fn populate_using_spks<I: IntoIterator<Item = ScriptBuf>>(
+        &self,
+        tx_graph: &mut TxGraph<ConfirmationTimeHeightAnchor>,
+        spks: I,
+        parallel_requests: usize,
+    ) -> Result<(), Error>
+    where
+        I::IntoIter: ExactSizeIterator,
+    {
+        self.populate_using_keychain_spks(
+            tx_graph,
+            spks.into_iter().enumerate().map(|(i, spk)| (i as u32, spk)),
+            usize::MAX,
+            parallel_requests,
+        )
+        .map(|_| ())
+    }
+
+    fn populate_using_txids<I: IntoIterator<Item = Txid>>(
```

Comment:
This is probably just a personal nit, but I'm of the opinion that the codebase would look "neater" if the naming scheme of conceptually similar methods were the same across both Electrum and Esplora, i.e. match `bdk_electrum`'s `populate_with_txids`.

Reply:
Thanks for the suggestion. The reason why I have a different name is because we aren't exactly populating the update graph with txids. Instead, we are using txids to fetch txs & anchors for the tx graph. However, I do feel the name can be improved. Let me know if you have any suggestions.

Reply:
I think I lean towards something similar to `chain_update`, maybe `graph_update_with_txids` or something similar? What do you think?

Reply:
it's not populating, it's fetching transactions from a server. The fact that it's doing IO and fertching txs should be emphasized.

Reply:
I mean technically the method is doing both? It's fetching txs and tx metadata from Esplora and then populating a `TxGraph` with it?

However, I'm thinking it's better to change the method signatures to **output** a `TxGraph` instead - this creates a nicer API, and the performance sacrifice is not too much. I had `&mut TxGraph` as an input because that was how the internal logic worked and it "felt more efficient" (which it is, but not by much).

Then we can change the methods to `fetch_txs_of_txids`, `fetch_txs_of_outpoints`, `fetch_txs_of_spks`, `fetch_txs_of_keychain_spks`, etc.

Reply:
Changed the methods to be as mentioned above. Except I've decided to use `with` as a preposition (instead of `of`).